search/includes/classes/class-healthjob.php

```diff
@@ -115,6 +124,42 @@ public function filter_cron_schedules( $schedule ) {
 		return $schedule;
 	}
 
+	/**
+	 * Check index health
+	 */
+	public function validate_contents() {
+		// Check if job has been disabled
+		if ( ! $this->is_enabled() ) {
+			$this->disable_job();
```

Comment:
We need to update `disable_job()` to also disable the new event.

Reply:
Nice catch! Done.